### PR TITLE
Fetch google calendar events

### PR DESCRIPTION
**Pull request by mayankagnihotri7**
## Notion Card
[Notion Card](https://www.notion.so/saeloun/As-a-Miru-user-I-should-be-able-to-integrate-my-account-with-Google-Calendar-01c06121941847509bdf34a4edd31340)

## Description
- This PR is a part of Google Calendar integration to Miru. In this, we fetch the events from Google Calendar API and send it as a JSON response to the frontend with only the required keys.

### CODE REVIEW

**Comment by mayankagnihotri7** [comment]
@gowsik-ragunath I've made the requested changes. Please review.

File: app/controllers/internal_api/v1/calendars_controller.rb [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  private
+
+    def client_options
+      {
+        client_id: ENV.fetch("GOOGLE_CLIENT_ID", ""),
+        client_secret: ENV.fetch("GOOGLE_CLIENT_SECRET", ""),
+        authorization_uri: "https://accounts.google.com/o/oauth2/auth",
+        token_credential_uri: "https://oauth2.googleapis.com/token",
+        scope: Google::Apis::CalendarV3::AUTH_CALENDAR_READONLY,
+        redirect_uri: "#{ENV.fetch("APP_BASE_URL", "")}/internal_api/v1/calendars/callback"
```
Comment (gowsik-ragunath) [inline app/controllers/internal_api/v1/calendars_controller.rb:68]: Can we use rails path instead like `internal_api_v1_callback_url`?

If you face `raise ArgumentError, "Missing host to link to! Please provide the :host parameter, set default_url_options[:host], or set :only_path to true"` please ref [this stackoverflow discussion](https://stackoverflow.com/a/5725343).

File: app/policies/calendar_policy.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    user_owner_role? || user_admin_role? || user_book_keeper_role? || user_client_role? || user_employee_role?
+  end
+
+  def calendars?
+    user_owner_role? || user_admin_role? || user_book_keeper_role? || user_client_role? || user_employee_role?
+  end
+
+  def events?
+    user_owner_role? || user_admin_role? || user_book_keeper_role? || user_client_role? || user_employee_role?
+  end
```
Comment (gowsik-ragunath) [inline app/policies/calendar_policy.rb:18]: As all the conditions are same, can we move this to a method and call that or we can directly return true.


```suggestion
  def redirect?
    allow_all_role?
  end

  def callback?
    allow_all_role?
  end

  def calendars?
    allow_all_role?
  end

  def events?
    allow_all_role?
  end
  
  def allow_all_role?
    user_owner_role? || user_admin_role? || user_book_keeper_role? || user_client_role? || user_employee_role?
  end
```

or


```suggestion
  def redirect?
    true
  end

  def callback?
    true
  end

  def calendars?
    true
  end

  def events?
    true
  end
```

**Review by gowsik-ragunath — APPROVED** [review]
LGTM! added a few minor comments

**Pull request merged**